drivers/gpu/drm/vc4/vc4_plane.c

```diff
@@ -520,6 +521,33 @@ static u32 vc4_lbm_size(struct drm_plane_state *state)
 	return lbm;
 }
 
+/*
+ *  Returns fractional offset of chroma with
+ *  0   = chroma cosited with luma
+ *  128 = chroma interstitial bewteen luma samples
+ *  256 = chroma cosited with second luma sample
+ */
+static int vc4_chroma_siting(enum drm_chroma_siting chroma_siting, bool is_horizontal)
+{
+	/* if unspecified use the most common format (MPEG2/H264) */
+	switch (chroma_siting) {
+		case DRM_CHROMA_SITING_LEFT:
+		case DRM_CHROMA_SITING_UNSPECIFIED:
+		default:
+			return is_horizontal ? 0:128;
+		case DRM_CHROMA_SITING_CENTER:
+			return is_horizontal ? 128:128;
+		case DRM_CHROMA_SITING_TOPLEFT:
+			return is_horizontal ? 0:0;
+		case DRM_CHROMA_SITING_TOP:
+			return is_horizontal ? 128:0;
+		case DRM_CHROMA_SITING_BOTTOMLEFT:
+			return is_horizontal ? 0:256;
+		case DRM_CHROMA_SITING_BOTTOM:
+			return is_horizontal ? 128:256;
+	}
```

Comment:
2D array of [DRM_CHROMA_SITING_MAX][2] instead of switch/case and then ternary expression?